launcher-implementation/src/main/scala/xsbt/boot/Launch.scala

```diff
@@ -304,9 +304,9 @@ class Launch private[xsbt] (val bootDirectory: File, val lockBoot: Boolean, val
       lazy val entryPoint: Class[T] forSome { type T } =
         {
           val c = Class.forName(id.mainClass, true, loader)
-          if (classOf[xsbti.AppMain].isAssignableFrom(c)) c
+          if (ServerApplication.isServerApplication(c)) c
```

Comment:
Do we need to change any documentation around this?   Also, do we have a priority check test for this switch?  Do we care?